lib/screens/login_screen.dart

```diff
@@ -0,0 +1,124 @@
+  LoginScreen({Key? key}) : super(key: key);
+
+  final _formkey = GlobalKey<FormState>();
+  String email = '';
+  String password = '';
+  bool isloading = false;
+  final FirebaseService _firebaseService = FirebaseService();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Form(
+        key: _formkey,
+        child: Container(
+          height: double.infinity,
+          width: double.infinity,
+          color: Colors.grey[200],
+          child: SingleChildScrollView(
+            padding: const EdgeInsets.symmetric(horizontal: 25, vertical: 120),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                const Text(
+                  "Sign In",
+                  style: TextStyle(
+                      fontSize: 50,
+                      color: kAppColor,
+                      fontWeight: FontWeight.bold),
+                ),
+                const SizedBox(height: 80),
+                TextFormField(
+                  keyboardType: TextInputType.emailAddress,
+                  onChanged: (value) {
+                    email = value;
+                  },
+                  validator: (value) {
+                    if (value!.isEmpty) {
+                      return "Please enter Email";
+                    }
+                  },
+                  textAlign: TextAlign.center,
+                  decoration: kTextFieldDecoration.copyWith(
+                    hintText: 'Email',
+                    prefixIcon: const Icon(
+                      FontAwesomeIcons.mailBulk,
+                      color: kAppColor,
+                    ),
+                  ),
+                ),
+                const SizedBox(height: 30),
+                TextFormField(
+                  obscureText: true,
+                  validator: (value) {
+                    if (value!.isEmpty) {
+                      return "Please enter Password";
+                    }
+                  },
+                  onChanged: (value) {
+                    password = value;
+                  },
+                  textAlign: TextAlign.center,
+                  decoration: kTextFieldDecoration.copyWith(
+                    hintText: 'Password',
+                    prefixIcon: const Icon(
+                      FontAwesomeIcons.userLock,
+                      color: kAppColor,
+                    ),
+                  ),
+                ),
+                const SizedBox(height: 30),
+                SizedBox(
+                  width: double.infinity,
+                  child: ElevatedButton(
+                    onPressed: () async {
+                      if (_formkey.currentState!.validate()) {
+                        try {
+                          await _firebaseService.loginEmailAndPassword(
+                              email, password);
+                          Navigator.pushReplacement(
```

Comment:
Don't think this 'Navigator.pushReplacement' is needed here. LandingScreen() is always listening for when a user is logged in/logged out. It automatically sees this and returns either the home_screen or login_screen. Thus, the LandingScreen does not need to be pushed again as it is already listening for changes.

Reply:
corrected